### PR TITLE
Role based protection implementation. Closes #113 

### LINKED ISSUE #113: Implement Role-Based Access Control for Feature Endpoints
## Description
Implement role-based access control in our application to restrict feature access based on user roles. Using Google OAuth for authentication, we need to define roles like Admin and User and control access to various features and endpoints based on these roles. If a user lacks the necessary role for a specific feature, the corresponding endpoints should return an error indicating insufficient permissions.

## Acceptance Criteria
1. **Role Definition**: Define at least two custom roles - Admin and User.
2. **Access Control**: Implement access control in backend services, allowing only authorized roles to access specific endpoints.
3. **Error Handling**: Ensure that endpoints return an appropriate error message (like 403 Forbidden) when a user with insufficient permissions attempts access.
4. **Secure Annotation**: Utilize the `[Authorized()]` attribute or equivalent in the backend to protect controller methods based on roles.
5. **Documentation**: Update all relevant documentation with details on role-based access control implementation and usage.
6. **Testing**: Include comprehensive tests to ensure role-based access control is functioning as expected across various scenarios.

### PR DESCRIPTION
**Pull request by roflmyrlok**
This branch covers issue #113 and adds role based access to endpoints.

### CODE REVIEW

**Review by YaroslavKSE — APPROVED** [review]
High quality code by Andrew Tate. Approved!

File: Identity/Identity.Core/UsersService.cs [hunk 1]
```diff
@@ -1,76 +1,93 @@
...
         
-        await _repository.ChangeUserAsync(email);
+		await _repository.ChangeUserAsync(email);
         
-    }
+	}
+
+	public async Task<(bool, bool)> GetUserExistAndAdmin(string email)
+	{
+		var tmp = await _repository.FindByEmailAsync(email);
```
Comment (Bardin08) [inline Identity/Identity.Core/UsersService.cs:80]: What's with naming?!

File: Identity/Identity.Core/UsersService.cs [hunk 2]
```diff
@@ -1,76 +1,93 @@
...
+		{
+			return (false, false);
+		}
+
+		if (!tmp.IsAdmin)
+		{
+			return (true, false);
+		}
+
+		return (true, true);
```
Comment (Bardin08) [inline Identity/Identity.Core/UsersService.cs:91]: That's horrible! Better return `bool?`

File: Identity/Identity.Data/Migrations/20231203171615_AuthChecks.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+        }
+
+        /// <inheritdoc />
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
```
Comment (Bardin08) [inline Identity/Identity.Data/Migrations/20231203171615_AuthChecks.cs:20]: Why it's empty and here?

File: Identity/Identity/EndpointHandlers.cs [hunk 4]
```diff
@@ -35,19 +36,31 @@ public static async Task<object> AddUser([FromServices] UsersService service, Us
         return new { Message = "User added successfully." };
     }
 
-    public static async Task<object> MakeAdmin([FromServices] UsersService service, string email, string password)
+    public static async Task<object> MakeAdmin([FromServices] UsersService service, HttpContext httpContext, string emailToMakeAdmin, string password)
```
Comment (Bardin08) [inline Identity/Identity/EndpointHandlers.cs:39]: `emailToPromote___` sound better for me

Comment (Bardin08) [reply Identity/Identity/EndpointHandlers.cs:39]: Also makes sense to wrap this to some kind of request model

**Review by Bardin08 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**